<launch>
    <param name="/use_sim_time" value="true" />

    <node pkg="tf2_ros" type="static_transform_publisher" name="base_to_front" args="0.35 0 0 0 0 0.5150381 0.8571673 base_link laser_front" />
    <node pkg="tf2_ros" type="static_transform_publisher" name="base_to_rear" args="-0.35 0 0 0 0 0.866 0.5 base_link laser_rear" />
    <node pkg="tf2_ros" type="static_transform_publisher" name="footprint_to_base" args="0 0 0 0 0 0 1 base_footprint base_link" />
    <node pkg="tf2_ros" type="static_transform_publisher" name="map_to_odom" args="0 0 0 0 0 0 1 map odom" />

    <node name="map_server" pkg="map_server" type="map_server" args="$(find project2)/maps/map.yaml" output="screen">
        <param name="frame_id" value="map" />
    </node>

    <!-- merge the two separate laser scans together, publish on topic /merge/scan -->
    <node pkg="ira_laser_tools" name="laserscan_multi_merger" type="laserscan_multi_merger" output="screen">
        <param name="destination_frame" value="base_link" />
        <param name="scan_destination_topic" value="/merge/scan" />
        <param name="laserscan_topics" value="/front/scan /rear/scan" />
    </node>

    <!-- broadcast the odometry as tf frame -->
    <node pkg="project2" type="broadcast_odom_tf" name="broadcast_odom_tf" output="screen">
        <param name="parent_frame" value="odom" />
        <param name="child_frame" value="base_link" />
        <param name="odom_topic" value="odom" />
    </node>

    <node pkg="project2" type="save_map_trajectory" name="save_map_trajectory" output="screen" />

    <node pkg="amcl" name="amcl" type="amcl">
        <remap from="scan" to="merge/scan" />
        <param name="use_map_topic" value="true" />
        <param name="odom_frame_id" value="odom" />
        <param name="base_frame_id" value="base_link" />
        <param name="global_frame_id" value="map" />
        <param name="initial_pose_x" value="0.0" />
        <param name="initial_pose_y" value="0.0" />
        <param name="initial_pose_a" value="0.0" />
        <param name="gui_publish_rate" value="10.0" />
        <param name="odom_model_type" value="omni" />
    </node>

    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find project2)/config/amcl.rviz" />

</launch>